lib/beaker/hypervisor/docker.rb

```diff
@@ -65,6 +65,19 @@ def provision
         host['docker_container'] = container
         host['docker_image'] = image
       end
+
+      hack_docker_etc_hosts @hosts, @options
+
+    end
+
+    def hack_docker_etc_hosts hosts, opts
+      etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
+      hosts.each do |host|
+        etc_hosts += "#{host['docker_container'].json["NetworkSettings"]["IPAddress"]}\t#{host.name}\n"
```

Comment:
Unfortunately not: this is always set to `0.0.0.0` if the `'DOCKER_HOST'` env_var isn't set - it get's set to the host machine's IP here => https://github.com/madAndroid/beaker/blob/hack_etc_hosts_docker/lib/beaker/hypervisor/docker.rb#L49 .. and AIUI, it get's set to the DOCKER_HOST's IP when that is set. This is to allow Beaker to connect via the port forward that Docker creates.